### PR TITLE
Update libddwaf to 1.5.1

### PR DESCRIPTION
**Pull request by CarlesDD**
### What does this PR do?

Updates libddwaf to 1.5.1

### Additional Notes

https://github.com/DataDog/libddwaf/blob/master/UPGRADING.md

### Checklist

- [ ] The CHANGELOG.md has been updated
- [ ] Unit tests have been updated and pass
- [ ] If known, an appropriate milestone has been selected

### CODE REVIEW

**Comment by simon-id** [comment]
Pretty sure we need the `ddwaf_update_rule_data` feature for ip blocking.

File: src/main.cpp [hunk 1]
```diff
@@ -241,14 +233,15 @@ Napi::Value DDWAFContext::run(const Napi::CallbackInfo& info) {
     mlog("Set total_runtime");
     res.Set("totalRuntime", Napi::Number::New(env, result.total_runtime));
   }
-  if (code != DDWAF_GOOD) {
+  if (code != DDWAF_OK) {
     res.Set("data", Napi::String::New(env, result.data));
   }
```
Comment (simon-id) [inline src/main.cpp:238]: I think `data` can be moved directly into the `DDWAF_MATCH` block
```suggestion
```

File: src/main.cpp [hunk 2]
```diff
@@ -241,14 +233,15 @@ Napi::Value DDWAFContext::run(const Napi::CallbackInfo& info) {
...
-  if (code == DDWAF_MONITOR) {
-    res.Set("action", Napi::String::New(env, "monitor"));
-  }
-  if (code == DDWAF_BLOCK) {
-    res.Set("action", Napi::String::New(env, "block"));
+  if (code == DDWAF_MATCH) {
+    res.Set("status", Napi::String::New(env, "match"));
```
Comment (simon-id) [inline src/main.cpp:240]:
```suggestion
    res.Set("status", Napi::String::New(env, "match"));
    res.Set("data", Napi::String::New(env, result.data));
```

**Comment by simon-id** [comment]
Also you missed a `action` -> `status` in the test file line 235

File: src/main.cpp [hunk 3]
```diff
@@ -241,14 +233,15 @@ Napi::Value DDWAFContext::run(const Napi::CallbackInfo& info) {
...
+    Napi::Array actions = Napi::Array::New(env, result.actions.size);
+    for (uint32_t i = 0; i < result.actions.size; ++i) {
```
Comment (uurien) [inline src/main.cpp:242]: `result.actions` has always a value? Can it be `nullptr` and create `SEGFAULT` here?

Comment (uurien) [reply src/main.cpp:240]: Yes, it looks fine

File: binding.gyp [hunk 4]
```diff
@@ -23,6 +23,7 @@
         'ldflags': ['-Wl,--rpath=\$$ORIGIN']
       }],
       ["OS == 'win'", {
+        "libraries": ["Ws2_32.lib"],
```
[the remark this answers is not in the document]
Comment (CarlesDD) [reply binding.gyp:26]: Windows build was failing due to some missing external symbols.

`ddwaf_static.lib(ip_utils.obj) : error LNK2001: unresolved external symbol __imp_inet_pton`

[the remark this answers is not in the document]
Comment (simon-id) [reply binding.gyp:26]: Right ok it's from the waf introducing the ip_utils thing make sense :+1:

File: test/index.js [hunk 5]
```diff
@@ -69,6 +70,43 @@ describe('DDWAF', () => {
...
+    ]
+
+    waf.updateRuleData(ruleData)
+    const result = context.run({ 'http.client_ip': IP_TO_BLOCK }, TIMEOUT)
+
+    assert.strictEqual(result.timeout, false)
+    assert.strictEqual(result.status, 'match')
+    assert(result.data)
+    assert(result.actions)
+    assert.strictEqual(result.actions[0], 'block')
```
Comment (simon-id) [inline test/index.js:94]:
```suggestion
    assert.deepStrictEqual(result.actions, ['block'])
```

**Review by simon-id — CHANGES_REQUESTED** [review]

**Review by simon-id — APPROVED** [review]

File: package.json [hunk 6]
```diff
@@ -1,9 +1,9 @@
 {
   "name": "@datadog/native-appsec",
-  "version": "1.3.0",
+  "version": "2.0.0",
```
[the remark this answers is not in the document]
Comment (simon-id) [reply package.json:3]: This is done automatically by `npm version major`

**Pull request merged**